interview-questions.md

```diff
@@ -8,48 +8,109 @@ Then, research the question to expand on your answer. Even if you feel you have
+
+1) Resolution, which is where yarn starts resolving dependencies by making request to the registry and recursively looking up each dependency.
+
+2) Fetching, which is where yarn looks in a global cache directory to see if the package needed has already been downloaded.If it hasnt, yarn fetches the tarball for the package and places it in the globall cache so it can work offline and wont need to download dependences more than once.
+
+3) Linking, in this step, yarn links everything together by copying all the files needed from the global cache into the local node modules directory.
+
+I personally think its incredible how fast this happens, I have installed yarn before, and have seen this steps be listed in the installment process, but little did I know how much was actually  going into this process. I just think its amazing.
+
 3. What is a React component?
 
-Your answer:
+Your answer: 
+
+React components are containers of reusable code. They can be stored in separate files in a components folder to keep organized and imported into the application.
+
+Researched answer: 
+
+Components come in 2 types and are independent and reusable bits of code. React components serve pretty much the same purpose as JavaScript function, but they work in isolation and return HTML. The 2 types of components are class components and function components.To elavorate a little more on each of these components, functional components can be easily explained by telling you that they are basically JavaScript functions. Function components is what I have been using and learning in the learning code journey. Class components can be considered to be more complex, this would be because class components can work with each other. Meaning, that with class components, data and information can be passed from one to another. Isnt that incredible?
+
+ I have learned that keeping components in separate files is very useful and keeps everything looking neater, but when keeping components in separate files, you must import the file in your application in order for it to work. This little tip has allowed me to keep good track of where everything is and has made my projects and challenges just look so much nicer.
```

Comment:
This is perfect! You are even bringing in a tidbit about having experience and why you understand it's importance. This is really good stuff Natally!